server/polar/customer_portal/schemas/order.py

```diff
@@ -61,3 +61,33 @@ class CustomerOrderInvoice(Schema):
 
 class CustomerOrderUpdate(OrderUpdateBase):
     """Schema to update an order."""
+
+
+class CustomerOrderPaymentStatus(Schema):
+    """Payment status for an order."""
+
+    status: str = Field(..., description="Current payment status.")
+    error: str | None | None = Field(
+        None, description="Error message if payment failed."
+    )
+
+
+class CustomerOrderConfirmPayment(Schema):
+    """Schema to confirm a retry payment using a Stripe confirmation token."""
+
+    confirmation_token_id: str = Field(
+        ..., description="ID of the Stripe confirmation token."
+    )
+
+
+class CustomerOrderPaymentConfirmation(Schema):
+    """Response after confirming a retry payment."""
+
+    status: str = Field(..., description="Payment status after confirmation.")
+    requires_action: bool = Field(
```

Comment:
Is it really needed? It looks like we just check `status == 'requires_action'`.

If you think that's useful, I would suggest to generate it as a computed field:

```py
class CustomerOrderPaymentConfirmation(Schema):
    ...
    
    @computed_field(description="Whether the payment requires additional action (3DS).")
    def requires_action(self) -> bool:
        return status == "requires_action"
```